kolibri/core/assets/src/api-resource.js

```diff
@@ -187,8 +190,12 @@ class Model {
     return promise;
   }
 
+  get orderedUrlParams() {
+    return this.resource.urlParams.map((key) => this.urlParams[key]);
+  }
+
   get url() {
-    return this.resource.modelUrl(this.id);
+    return this.resource.modelUrl(...this.orderedUrlParams, this.id);
```

Comment:
Unfortunately, we can't - I would love to be able to just pass in the kwargs too - but that would require a rewrite of Django JS reverse to treat kwarg URLs differently from arg URLs, which it does not currently.

This will work for now, and I can look at submitting an upstream change to Django JS Reverse in the future.